libkbfs/tlf_journal.go

```diff
@@ -809,6 +808,62 @@ func (j *tlfJournal) getJournalStatus() (TLFJournalStatus, error) {
 	}, nil
 }
 
+func (j *tlfJournal) getJournalStatus() (TLFJournalStatus, error) {
+	j.journalLock.RLock()
+	defer j.journalLock.RUnlock()
+	return j.getBaseJournalStatusLocked()
+}
+
+func (j *tlfJournal) getJournalStatusWithPaths(ctx context.Context,
+	config Config, mdOps journalMDOps, blocks *folderBlockOps) (
+	TLFJournalStatus, error) {
+	j.journalLock.RLock()
+	defer j.journalLock.RUnlock()
+	status, err := j.getBaseJournalStatusLocked()
+	if err != nil {
+		return TLFJournalStatus{}, err
+	}
+
+	// Make chains over the entire range to get the unflushed files.
+	// TODO: cache this chains object and update it when new revisions
+	// appear and when revisions are flushed, instead of rebuilding
+	// every time.
+	if status.RevisionEnd != MetadataRevisionUninitialized {
+		var irmds []ImmutableRootMetadata
+		// These GetRange calls end up taking journalLock for reading
```

Comment:
Another option would be to fix up `decryptMDPrivateData` to take as parameters whatever it needs from `Config` instead of just the whole thing. It looks like it just needs `codec`, `crypto`, `encryptionKeyGetter` can be extended to have `GetTLFCryptKeyForMDDecryption` also, a `uid` (which `tlfJournal` already has), `BlockCache`, and `BlockOps`. If `tlfJournalConfig` is extended to have `BlockOps`, I think that covers it. Maybe we could keep `decryptMDPrivateData` as is and split off the other code into a helper function so that you don't have to fix up all the other call sites?

I'm just wary of having `tlfJournal` depend on everything in `Config` -- just thinking of the future when we might want to split it off into another package.